File: 78-subsets/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        def myRec(arr,level,res):
            if level == len(nums):
                res.append(arr[:])
                return
            myRec(arr,level + 1,res)
            arr.append(nums[level])
            myRec(arr, level + 1,res)
            arr.pop()
            return
        res = []
        myRec([],0,res)
        return res

File: 78-subsets/subsets.java
class Solution {
    public List<List<Integer>> result;
    public int [] numsG;
    public List<List<Integer>> subsets(int[] nums) {
        result = new ArrayList<>();
        numsG = nums;
        fun(0,new ArrayList<>());
        return result;
    }
    public void fun(int index, List<Integer> curr){
        if(index == numsG.length){
            result.add(new ArrayList<>(curr));
            return;
        }
        curr.add(numsG[index]);
        fun(index + 1, curr);
        curr.remove(curr.size()-1);
        fun(index + 1, curr);
    }
}